llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -63,12 +63,12 @@ MachineInstr *findPreIncMatch(MachineInstr &MemI, MachineRegisterInfo &MRI,
   return nullptr;
 }
 
-/// Checks if any operand of \a Use is defined by \a MI.
-/// This is not transitive: it will not look at how the uses of \a MI are
+/// Checks if any operand of \a MI is defined by \a Def.
+/// This is not transitive: it will not look at how the uses of \a Def are
 /// defined.
-bool isUseOf(const MachineInstr &MI, const MachineInstr &Use) {
-  for (auto &Defs : Use.defs()) {
-    for (auto &MIUse : MI.uses()) {
+bool isUseOf(const MachineInstr &MI, const MachineInstr &Def) {
+  for (auto &Defs : Def.all_defs()) {
+    for (auto &MIUse : MI.all_uses()) {
```

Comment:
Could you add a test to check that the `VPACK`/`VUNPACK` isn't advanced when one of its implicit inputs is defined in between the load/store and the op?
E.g.
```
%0 = VLD ...
VST ... ; prevent delaying VLD
$crSign = ...
%1 = VPACK %0, implicit $crSign
```

Reply:
I included a test for VUNPACK with a control register write in between. I am also looking explicitly to `G_INTRINSIC_W_SIDE_EFFECTS` when we try to advance an instruction.